[#453] package rollup binaries (#519)
serokell/tezos-packaging: docs/tx_rollup.md

<!--
   - SPDX-FileCopyrightText: 2022 Oxhead Alpha
   - SPDX-License-Identifier: LicenseRef-MIT-OA
   -->

> 

Reviewer: IMO it would be nice to mention this document in the README, maybe close by to the link to the baking doc

# Setting up transaction rollup node on Ubuntu

`proto` variable can hold every active protocol, currently `013-PtJakart` and `014-PtKathma`.

> By default, `tezos-tx-rollup-node-$proto.service` will using:

Reviewer: ```suggestion
By default, `tezos-tx-rollup-node-$proto.service` will use:
```

* `/var/lib/tezos/.tezos-client` as the `tezos-client` data directory, set in `DATA_DIR`
* `http://localhost:8732` as the `tezos-node` RPC address, set in `NODE_RPC_ENDPOINT`
* `127.0.0.1:8472` as the `tezos-tx-rollup-node-$proto` RPC address, set in `ROLLUP_NODE_RPC_ENDPOINT`

> * `observer` as the `tezos-tx-rollup-node-$proto` working mode, set in `MODE`

Reviewer 2: ```suggestion
* `observer` as the `tezos-tx-rollup-node-$proto` working mode, set in `MODE`
```
```suggestion
* `observer` as the `tezos-tx-rollup-node-$proto` working mode, set in `ROLLUP_MODE`
```

* `rollup` as the rollup alias, set in `ROLLUP_ALIAS`

> In order to change defaults, edit `/etc/default/tezos-tx-rollup-node-$proto` file.

Reviewer 2: Let's mention that one needs to restart the service once the config file is updated

> In order to change defaults, edit `/etc/default/tezos-tx-rollup-node-$proto` file and restart the service.

Reviewer: ```suggestion
In order to change the defaults, edit `/etc/default/tezos-tx-rollup-node-$proto` and restart the service.
```

> 

Reviewer: Perhaps it's worth giving an example of how to do the edit and the restart, see the baking doc for example.

At first you should originate rollup with the provided alias for some implicit account:
```

> tezos-client originate tx rollup ${ROLLUP_ALIAS} from ${IMPLICIT_ACCOUNT_ALIAS}

Reviewer 2: I suppose this command should be run under `tezos` user to share the alias with the services
```suggestion
sudo -u tezos tezos-client originate tx rollup ${ROLLUP_ALIAS} from ${IMPLICIT_ACCOUNT_ALIAS}
```

> sudo -u tezos tezos-client originate tx rollup ${ROLLUP_ALIAS} from ${IMPLICIT_ACCOUNT_ALIAS}

Reviewer: I suppose that instead of the syntax for bash inlining, e.g. `${ROLLUP_ALIAS}` here we should instead be consistent with the other docs and use angular brackets, e.g. `<ROLLUP_ALIAS>`

```

After that, with the active `tezos-node` service available with the provided uri, run the following command
```
sudo systemctl start tezos-tx-rollup-node-$proto.service
```

> For futher details, see [documentation](https://gitlab.com/tezos/tezos/-/blob/master/docs/jakarta/transaction_rollups.rst).

Reviewer: I would change this link to:
```suggestion
For futher details, see [the documentation](http://tezos.gitlab.io/active/transaction_rollups.html).
```
because this one is rendered in a webpage, but also because point at the `active` protocol, which is updated over time.
